src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -1503,36 +1477,41 @@ public Set getGroupMembershipRoles(String userDn, String username) {
                     String role = ga.getAuthority();
 
                     // backward compatible name mangling
-                    if (convertToUpperCase)
+                    if (isConvertToUpperCase())
                         role = role.toUpperCase();
-                    r.add(new GrantedAuthorityImpl(rolePrefix + role));
+                    r.add(new SimpleGrantedAuthority(getRolePrefix() + role));
                 }
             }
 
             return r;
         }
 
         public boolean isGeneratingPrefixRoles() {
-            return StringUtils.isNotBlank(rolePrefix) || convertToUpperCase;
+            return StringUtils.isNotBlank(getRolePrefix()) || isConvertToUpperCase();
         }
 
-        public boolean isConvertToUpperCase() {
-            return convertToUpperCase;
-        }
+        // Needed by FromGroupSearchLDAPGroupMembershipStrategy:
 
-        public String getRolePrefix() {
-            return rolePrefix;
+        private GrantedAuthority defaultRole;
+
+        @Override
+        public void setDefaultRole(String defaultRole) {
```

Comment:
Awkwardly, the supertype has

```java
	/**
	 * Returns the default role Method available so that classes extending this can
	 * override
	 * @return the default role used
	 * @see #setDefaultRole(String)
	 */
	private GrantedAuthority getDefaultRole() {
		return this.defaultRole;
	}
```

which does not make sense—it is _not_ accessible.